### PR TITLE
account for user.state = incomplete_signup in a few more places

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner _might_ be solving the problem in `KifiSiteRouter`. I can add a guard on auto-joining, but I think it's best to throw an airbrake so we can figure out when/why this is happening.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala [hunk 1]
```diff
@@ -410,6 +411,7 @@ class AuthHelper @Inject() (
     authCommander.finalizeEmailPassAccount(efi, request.userId, request.user.externalId, request.identityOpt, inviteExtIdOpt).map {
       case (user, email, newIdentity) =>
         val verifiedEmail = verifySignupEmail(request.userId, email, libraryPublicId, libAuthToken)
+        if (user.state == UserStates.INCOMPLETE_SIGNUP) airbrake.notify(s"[finalizeEmailPassAccount] calling finishSignup without an active user! ${user.toString}")
         finishSignup(user, email, newIdentity, emailConfirmedAlready = verifiedEmail, libraryPublicId = libraryPublicId, libAuthToken = libAuthToken, orgPublicId, orgAuthToken, isFinalizedImmediately = false)
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:413]: This state might be normal at this point, before `finishSignup` @andrewconner ?

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:413]: We had an org membership where the user's state was `incomplete_signup`, which caused bugs, so this is an attempt to figure out how people are being created with such a state. There's a number of them keeping, joining libraries, etc.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:413]: @leogrim you should be right that by this point things would be good. I had assumed that the user was auto-joined into the org, but I see now that he got a user-invitation. So I can remove this log, the issue isn't that we're auto-joining people with incomplete signup, it's that we're letting them do anything at all.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:413]: > this is an attempt to figure out how people are being created with such a state

So sorry! I wasn't clear earlier. I know how it happens (or at least, how it used to happen?), those users never go through this code path, so likely are broken in other ways as well. It happens if a user abandons registration mid-signup using email, and goes straight to kifi.com. Your other fix is solid, it'll do it.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:413]: I just realized that, will take out the extra stuff.

**Comment by camhashemi** [comment]
@andrewconner another thing I added was to use `user.state != ACTIVE` instead of `user.state == INACTIVE`, since there are more non-active states than `INACTIVE`. This could lead to broken experiences for users with non-active states (like the user who was a member of the org but didn't show up). Is that the right way to go about it in general? Or let them slide?

**Comment by andrewconner** [comment]
So it looks like sometime, BLOCKED stopped being honored. Sigh. Yeah you can redirect if `user.state != ACTIVE`, but for blocked/inactive, we probably should delete their cookies (log them out).

**Comment by camhashemi** [comment]
I'll add that logic (assuming that they won't be allowed to log back in), but I was referring to backend processing more than redirect logic.

 We take for granted that usually `state = active || inactive`. With users should we try to use `state != active` instead of `state == inactive`? this could break things for these non-active users.

**Pull request merged**